python/paddle/v2/framework/tests/test_mul_op.py

```diff
@@ -16,16 +16,37 @@ def setUp(self):
         self.outputs = {'Out': np.dot(self.inputs['X'], self.inputs['Y'])}
 
 
-class MulGradOpTest(GradientChecker):
-    def test_mul(self):
-        op = create_op("mul")
-        inputs = {
+class TestMulGradOp(GradientChecker):
+    def setUp(self):
+        self.op = create_op("mul")
+        self.inputs = {
             'X': np.random.random((32, 84)).astype("float32"),
             'Y': np.random.random((84, 100)).astype("float32")
         }
+
+    def test_cpu_gpu_compare(self):
+        self.compare_grad(self.op, self.inputs)
+
+    def test_normal(self):
         # mul op will enlarge the relative error
         self.check_grad(
-            op, inputs, set(["X", "Y"]), "Out", max_relative_error=0.5)
+            self.op, self.inputs, ["X", "Y"], "Out", max_relative_error=0.5)
+
+    def test_ignore_x(self):
+        self.check_grad(
+            self.op,
+            self.inputs, ["Y"],
+            "Out",
+            max_relative_error=0.5,
+            no_grad_set={"X"})
+
+    def test_ignore_y(self):
+        self.check_grad(
+            self.op,
+            self.inputs, ["X"],
+            "Out",
+            max_relative_error=0.5,
+            no_grad_set={"Y"})
```

Comment:
@qingqing01 The framework should automatically test cases with certain input gradient ignored without explicitly writing code here. We should change GradientChecker.check_grad() to do this.

Reply:
@emailweixu Thanks for your comments. Ok, in the first comments:  https://github.com/PaddlePaddle/Paddle/pull/3857/commits/447033296d927dd0b0c1240e2ecccaa667eb0fe8

The `check_grad` will automatically test. I'll change it again :).

Reply:
> The framework should automatically test cases with certain input gradient ignored without explicitly writing code here. We should change GradientChecker.check_grad() to do this.

* To make users invoke check_grad by themselves can handle if some IGs are not able to be ignored. For example, OpA has two inputs, 'X' and 'Y' and gradient 'Y' should always be calculated, and it YG should never be nullptr.
  * Also, `check_grad` is not only used to check single operator's gradient but also the gradient inside a whole neural network. It is hard to automatically testing a whole network.
* To make users invoke check_grad can make error log clearer. The error would be `test_mul_grad_ingore_x error`.
* To make user-interface easily, there maybe a parameter in `check_grad`, like `test_no_input_grad`. If `test_no_input_grad=True`, multiple `check_grad`s will be invoked.